jax_galsim/gsparams.py

```diff
@@ -0,0 +1,123 @@
+import copy
+from dataclasses import dataclass
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+
+
+@_wraps(_galsim.GSParams)
+@dataclass(frozen=True, repr=False)
+class GSParams:
+
+    minimum_fft_size: int = 128
+    maximum_fft_size: int = 8192
+    folding_threshold: float = 5.0e-3
+    stepk_minimum_hlr: float = 5
+    maxk_threshold: float = 1.0e-3
+    kvalue_accuracy: float = 1.0e-5
+    xvalue_accuracy: float = 1.0e-5
+    table_spacing: int = 1
+    realspace_relerr: float = 1.0e-4
+    realspace_abserr: float = 1.0e-6
+    integration_relerr: float = 1.0e-6
+    integration_abserr: float = 1.0e-8
+    shoot_accuracy: float = 1.0e-5
+
+    @staticmethod
+    def check(gsparams, default=None, **kwargs):
+        """Checks that gsparams is either a valid GSParams instance or None.
+
+        In the former case, it returns gsparams, in the latter it returns default
+        (GSParams.default if no other default specified).
+        """
+        if gsparams is None:
+            gsparams = default if default is not None else GSParams.default
+        elif not isinstance(gsparams, GSParams):
+            raise TypeError("Invalid GSParams: %s" % gsparams)
+        return gsparams.withParams(**kwargs)
+
+    def withParams(self, **kwargs):
+        """Return a `GSParams` that is identical to the current one except for any keyword
+        arguments given here, which supersede the current value.
+        """
+        if len(kwargs) == 0:
+            return self
+        else:
+            ret = copy.copy(self)
+            for k in kwargs:
+                if not hasattr(ret, "_" + k):
+                    raise TypeError("parameter %s is invalid" % k)
+                setattr(ret, "_" + k, kwargs[k])
+            return ret
+
+    @staticmethod
+    def combine(gsp_list):
+        """Combine a list of `GSParams` instances using the most restrictive parameter from each.
+
+        Uses the minimum value for most parameters. For the following parameters, it uses the
+        maximum numerical value: minimum_fft_size, maximum_fft_size, stepk_minimum_hlr.
+        """
+        if len(gsp_list) == 1:
+            return gsp_list[0]
+        elif all(g == gsp_list[0] for g in gsp_list[1:]):
+            return gsp_list[0]
+        else:
+            return GSParams(
+                max([g.minimum_fft_size for g in gsp_list if g is not None]),
+                max([g.maximum_fft_size for g in gsp_list if g is not None]),
```

Comment:
hummm, good question. the documentation says:
```
    Parameters:
        minimum_fft_size:   The minimum size of any FFT that may need to be performed.
                            [default: 128]
        maximum_fft_size:   The maximum allowed size of an image for performing an FFT.  This
                            is more about memory use than accuracy.  We have this maximum
                            value to help prevent the user from accidentally trying to perform
                            an extremely large FFT that crashes the program. Instead, GalSim
                            will raise an exception indicating that the image is too large,
                            which is often a sign of an error in the user's code. However, if
                            you have the memory to handle it, you can raise this limit to
                            allow the calculation to happen. [default: 8192]
```
so I think the one that matters the most is the minimum fft size as it controls accuracy, and it kinda makes sense to use the maximum one (because that's the most conservative one), for the maximum fft size... I could image both options might be ok, this one is only there to avoid blowing up the RAM.

Reply:
cool, then yeah it doesn't seem too important and I'm ok with `max`